sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client_async.py

```diff
@@ -0,0 +1,66 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import functools
+import os
+import pytest
+import six
+
+from devtools_testutils import AzureTestCase, PowerShellPreparer
+
+from azure.containerregistry import (
+    DeletedRepositoryResult,
+    RepositoryProperties,
+)
+from azure.containerregistry.aio import ContainerRegistryClient, ContainerRepositoryClient
+from azure.core.exceptions import ResourceNotFoundError
+from azure.core.paging import ItemPaged
+from azure.identity.aio import DefaultAzureCredential
+
+from asynctestcase import AsyncContainerRegistryTestClass
+
+
+acr_preparer = functools.partial(
+    PowerShellPreparer,
+    "containerregistry",
+    containerregistry_baseurl="fake_url.azurecr.io",
+)
+
+
+class TestContainerRegistryClient(AsyncContainerRegistryTestClass):
+
+    @acr_preparer()
+    async def test_list_repositories(self, containerregistry_baseurl):
+        client = self.create_registry_client(containerregistry_baseurl)
+
+        repositories = client.list_repositories()
+
+        count = 0
+        prev = None
+        async for repo in client.list_repositories():
+            count += 1
+            assert isinstance(repo, six.string_types)
+            assert prev != repo
```

Comment:
is there a concern for the service returning dup items?

Reply:
No, wanted to verify the deserialization.